server/parser.go

```diff
@@ -47,6 +47,7 @@ type pubArg struct {
 	queues  [][]byte
 	size    int
 	hdr     int
+	topic   []byte // This is MQTT published topic but is not restricted to MQTT producers!
```

Comment:
Isn't there a fixed way to go between the two?

Reply:
Not sure what you mean here. The comment was supposed to explain that although `topic` is used for the "subject" that is passed to the mqtt subscriptions, it can be used by non MQTT producers.
The case is a NATS publisher that has matching MQTT subscriptions. c.pa.topic will be empty but the first time we get to a mqttDeliverMsgCb(), we will do the conversion from subject to topic, but only once regardless of how many matching MQTT subscriptions there are for this NATS message.

Reply:
Meaning there is a well known way to get from MQTT topic to NATS subject and back, why do we need to store it? strings.Replace() may not be too fast but wondering if saving makes a noticeable difference.

Reply:
Unfortunately, it is not a simple strings.Replace(). There are more complex rules due to the how `/` can be used in MQTT. For instance, this is a valid MQTT topic: `/foo//bar/`, which a straight strings.Replace() would create an invalid NATS subject.

That being said, I am fine removing c.pa.topic and have the subject-to-topic conversion being made every time for NATS->MQTT but also (unfortunately) from MQTT QoS1 -> MQTT QoS1. The one case I can store in the specific MQTT producer's struct is MQTT Qos0.

Based on your response, I will update the PR (I am going to push an update anyway since I am adding some cases to the conversion test).

Reply:
Let's do that and mark it with comment around potential future optimization.